futures/src/lib.rs

```diff
@@ -237,6 +237,12 @@ pub mod io {
     };
 }
 
+#[cfg(feature = "std")]
+pub mod lock {
+    //! Futures-powered synchronization primitives.
+    pub use futures_util::lock::{Mutex, MutexAcquire, MutexGuard};
```

Comment:
Not sure how I feel about the `MutexAcquire` name. It would be perfect if the `lock()` method was named `acquire()`. What do you think about `MutexLock`?

Reply:
I understand wanting to name it after the method, but IMO `MutexLock` kind of sounds like you're holding the lock, which isn't right... Hm. Maybe we should just go ahead and call it `MutexLockFuture`? We don't typically include `Future` in the names of these types, but it could be helpful.

Reply:
Changed to `MutexLockFuture`.